backend/quotes/quotes-manager.js

```diff
@@ -57,22 +70,33 @@ function setQuoteIdIncrementer(number) {
 
 function addQuote(quote) {
     return new Promise(async (resolve, reject) => {
-
-        let newQuoteId = await getNextQuoteId();
-        if (newQuoteId == null) {
-            logger.error("Unable to add quote as we could not generate a new ID");
-            return reject();
+        // If no/invalid ID is specified, get the next one
+        if (!quote._id || isNaN(quote._id)) {
+            let newQuoteId = await getNextQuoteId();
```

Comment:
Looks like this `let` never gets reassigned and so could be a `const` instead

Reply:
Fixed